### PR TITLE
Feature/arm requirements extended

### PR DESCRIPTION
**Pull request by LarsJanssenTUe**

### CODE REVIEW

**Comment by MatthijsBurgh** [comment]
@LarsJanssenTUe conflicts from the start 👍🏻

**Comment by LarsJanssenTUe** [comment]
> @LarsJanssenTUe conflicts from the start 👍🏻

@MatthijsBurgh should be good now!

**Review by MatthijsBurgh — APPROVED** [review]
Wait for second approval before merging

**Review by PetervDooren — DISMISSED** [review]
Couple of comments. Over all I think it looks good.
Conceptually it does not sit right with me that arm requirements are specified for the statemachines rather than the individual arms used by the statemachines but I think that will be a topic for another day.

File: robot_skills/src/robot_skills/arm/handover_detector.py [hunk 1]
```diff
@@ -20,45 +21,50 @@ def __init__(self, robot_name, tf_listener, side):
         super(HandoverDetector, self).__init__(robot_name=robot_name, tf_listener=tf_listener)
         self.side = side
 
-    def handover_to_human(self, timeout=10):
+    def _generic_handover(self, direction, timeout=10):
         """
-        Handover an item from the gripper to a human.
-
+        Handover an item to/from the gripper from/to a human.
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/arm/handover_detector.py]: What do you think of this one. Small change, but enough change IMO.

File: challenge_cleanup/src/challenge_cleanup/self_cleanup.py [hunk 2]
```diff
@@ -331,17 +327,14 @@ def __init__(self, robot, selected_entity_designator, room_des):
                                    transitions={"done": "PLACE_IN_TRASH",
                                                 "failed": "SAY_PLACE_FAILED"})
 
-            arm_properties_place = {"required_trajectories": ["prepare_place"],
-                                    "required_gripper_types": [arms.GripperTypes.GRASPING]}
-            arm_designator_place = ds.OccupiedArmDesignator(robot, arm_properties_place, name="occupied_arm_designator")
+            arm_designator_place = ds.OccupiedArmDesignator(robot, name="occupied_arm_designator").lockable()
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply challenge_cleanup/src/challenge_cleanup/self_cleanup.py:330]: True. Normally each pick-up and place operation should be done with the same arm, though the same designator instance. Other combo operations, can be done with another arm. Though another designator.

But indeed in this situation one designator should be used.

File: robot_skills/src/robot_skills/robot.py [hunk 3]
```diff
@@ -1,9 +1,15 @@
 #! /usr/bin/env python
 
-from collections import OrderedDict, Sequence
+from collections import OrderedDict
+try:
+    # Python 3
+    from collections.abc import Sequence, Set
+except ImportError:
+    # Python 2
+    from collections import Sequence, Set
```
[the remark this answers is not in the document]
Comment (MatthijsBurgh) [reply robot_skills/src/robot_skills/robot.py:9]: I think this is a good pratice.

**Review by MatthijsBurgh — DISMISSED** [review]
See https://github.com/tue-robotics/tue_robocup/pull/1092/files/0f5dd247a3b6b46855a49240d2b53bdfc98d12f9#r615438247

**Review by PetervDooren — APPROVED** [review]
Tentative approve. I cant clearly see the changes compared to master

**Pull request merged**